lblod/ember-rdfa-editor
# GN-4568: Fallback to "align" attribute when parsing DOM

### Overview

GN-4568: Fallback to "align" attribute when parsing DOM.

Editor will attempt to use the `align` attribute if the `text-align` property of `style` attribute is not present to determine the alignment when parsing DOM.

##### connected issues and PRs:

https://binnenland.atlassian.net/browse/GN-4658

### Setup

1. Checkout
2. `npm run start` 

### How to test/reproduce

1. Open editor
2. Open console
3. Paste below code into console (contains `p` tags with `align` attributes instead of `text-align` in `style`)
4. Observe following layout

<img width="697" alt="CleanShot 2024-01-03 at 11 13 54@2x" src="https://github.com/lblod/ember-rdfa-editor/assets/769698/e11101b2-4584-4fa9-8774-1b79281f0798">


<details>
<summary>Paste code</summary>

```ts
const data = `<html
  xmlns:o="urn:schemas-microsoft-com:office:office"
  xmlns:w="urn:schemas-microsoft-com:office:word"
  xmlns:m="http://schemas.microsoft.com/office/2004/12/omml"
  xmlns="http://www.w3.org/TR/REC-html40"
>
  <head>
    <meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
    <meta name="ProgId" content="Word.Document" />
    <meta name="Generator" content="Microsoft Word 15" />
    <meta name="Originator" content="Microsoft Word 15" />
  </head>

  <body lang="EN-IE" style="tab-interval: 36pt; word-wrap: break-word">
    

    <p class="MsoNormal">
      <span lang="LV" style="mso-ansi-language: LV"
        >Left aligned paragraph.</span
      >
    </p>

    <p class="MsoNormal" align="center">
      <span lang="LV" style="mso-ansi-language: LV"
        >Center aligned paragraph.</span
      >
    </p>

    <p class="MsoNormal" align="right">
      <span lang="LV" style="mso-ansi-language: LV"
        >Right aligned paragraph.</span
      >
    </p>

    <p class="MsoNormal" align="right">
      <span lang="LV" style="mso-ansi-language: LV"></span>
    </p>

    <p class="MsoNormal" align="justify">
      <span lang="LV" style="mso-ansi-language: LV"
        >Justified paragraph. Ullamco ex consequat ipsum enim id sit mollit
        culpa deserunt non ut do reprehenderit. In ad dolor voluptate proident
        non aliqua sit ullamco. Id culpa sit exercitation sunt est ullamco
        dolore nostrud proident veniam dolor. In aute adipisicing laboris.
        Exercitation qui sunt reprehenderit occaecat commodo non minim ea duis
        deserunt.</span
      >
    </p>

    <p class="MsoNormal" align="center" style="text-align: center">
      <span lang="LV" style="mso-ansi-language: LV"></span>
    </p>

    <p class="MsoNormal">
      <span lang="LV" style="mso-ansi-language: LV"></span>
    </p>

    
  </body>
</html>

`;

const clipboardEvent = new ClipboardEvent("paste", {
  dataType: "text/html",
  clipboardData: new DataTransfer(),
});

// clipboardEvent.clipboardData.setData("text/rtf", data);
clipboardEvent.clipboardData.setData("text/html", data);

const selector = ".say-editor__paper div.say-content";

document.querySelector(selector).dispatchEvent(clipboardEvent);

const clipboardEvent = new ClipboardEvent("paste", {
  dataType: "text/html",
  clipboardData: new DataTransfer()
});

// clipboardEvent.clipboardData.setData("text/rtf", data);
clipboardEvent.clipboardData.setData("text/html", data);

const selector = ".say-editor__paper div.say-content"

document.querySelector(selector).dispatchEvent(clipboardEvent);

```
</details>


### Challenges/uncertainties

We don't know what content was actually pasted by the QA reporter on the issue, so this fix is an _assumption_ on what went wrong. The fix is also a good fallback strategy, IMO, regardless of what was the actual issue.

I've asked QA to provide the clipboard content here - https://binnenland.atlassian.net/browse/GN-4658?focusedCommentId=94824, maybe we will be able to figure out some extra information.



### Checks PR readiness
